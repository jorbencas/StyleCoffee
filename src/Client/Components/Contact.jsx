import React from 'react';
import { contactService } from '../services/services';

class Contact extends React.Component {
    constructor(props){
        super(props);

        this.state = {
            name:'',
            message:'',
            email:'',
            subject:''
        }; 
        this.handleInputChange = this.handleInputChange.bind(this); 
        this.handleSubmit = this.handleSubmit.bind(this); 
    };   

    /*
    handleInputChange = (event) => {
        event.preventDefault();
        const target = event.target;
        const value = target.type === 'checkbox' ? target.checked : target.value;
        const name = target.name;
    
        this.setState({
          [name]: value
        });

        console.log(this.state);
    };
*/

handleInputChange(event) {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;
   
    this.setState({
      [name]: value
    });

    console.log(this.state);
}

handleSubmit(event) {
    event.preventDefault();
    contactService(this.state);
}

    render() {               
        return (
            <div className="grid-main">
                <div className="Contact">
                    <form id="contact_form" name="contact_form" className="form-contact">
                        <h1 className="contact_item">Contacto</h1>
                        <div className="contact_item">
                            <label htmlFor="name">name</label><br/>
                            <input required type="text" id="name" name="name" placeholder="Nombre" onChange={this.handleInputChange}/>
                        </div>
                        <div className="contact_item">
                            <label htmlFor="email">Email</label><br/>
                            <input required type="email" id="email" name="email" placeholder="Email *" onChange={this.handleInputChange}/>
                        </div>
                        <div className="contact_item">
                            <label className="inputSubject" htmlFor="inputSubject">Tema de Consulta</label><br/>
                            <select className="" id="inputSubject" name="subject" title="Choose subject" onChange={this.handleInputChange}>
                                <option value="">- Por favor, seleccione un tema de consulta -</option>
                                <option value="actividad">Info relativa a alguna actividad</option>
                                <option value="dpto">Contacta con nuestro dpto de actividades</option>
                                <option value="trabaja">Trabaja con nosotros</option>
                                <option value="sugerencias">Haznos sugerencias</option>
                                <option value="reclamaciones">Atendemos tus reclamaciones</option>
                                <option value="novedades">Te avisamos de nuestras novedades</option>
                                <option value="diferente">Algo diferente</option>
                            </select>
                        </div>
                        <div className="contact_item">
                            <label htmlFor="recomendation">Hace Aqui tu recomendaci√≥n</label><br/>
                            <textarea required className="contact_item" id="recomendation" name="message" placeholder="Introduzca aqui su mensaje *" onChange={this.handleInputChange}></textarea>
                        </div>
                        <div className="contact_item">
                            <input className="contact_Item" type="submit" name="submit" id="submit" value="Enviar"  onClick={this.handleSubmit}/>
                        </div>
                    </form>
                </div> 
            </div>
        );
    }
}

export default Contact;