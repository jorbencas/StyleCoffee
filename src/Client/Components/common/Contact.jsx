import React from 'react';
import { contactService } from '../../services/services';
import { FormErrors } from '../../lib/FormErrors';

class Contact extends React.Component {
    constructor(props){
        super(props);

        this.state = {
            name:'',
            message:'',
            email:'',
            subject:'',
            formErrors: {email: ''},
            emailValid: false,
            passwordValid: false,
            formValid: false
        }; 
        this.handleInputChange = this.handleInputChange.bind(this); 
        this.handleSubmit = this.handleSubmit.bind(this); 
        this.validateField = this.validateField.bind(this);
        this.validateForm = this.validateForm.bind(this);
    };   

handleInputChange(event) {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    this.setState({[name]: value}, () => { this.validateField(name, value) });

    console.log(this.state);
}

validateField(fieldName, value) {
    let fieldValidationErrors = this.state.formErrors;
    let emailValid = this.state.emailValid;
    let passwordValid = this.state.passwordValid;
  
    switch(fieldName) {
      case 'email':
        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
        fieldValidationErrors.email = emailValid ? '' : ' is invalid';
        break;
      case 'password':
        passwordValid = value.length >= 6;
        fieldValidationErrors.password = passwordValid ? '': ' is too short';
        break;
      default:
        break;
    }
    this.setState({formErrors: fieldValidationErrors,
                    emailValid: emailValid,
                    passwordValid: passwordValid
                  }, this.validateForm);
  }
  
  validateForm() {
    this.setState({formValid: this.state.emailValid && this.state.passwordValid});
  }

  errorClass(error) {
    return(error.length === 0 ? 'Tu debes escribir algo' : 'has-error');
  }

  handleSubmit(event) {
    event.preventDefault();
    contactService(this.state);
}


    render() {               
        return (
            <div className="grid-main">
                <div className="Contact">
                    <form id="contact_form" name="contact_form" className="form-contact">
                        <h1 className="contact_item">Contacto</h1>
                        <div><FormErrors formErrors={this.state.formErrors} /></div>
                        <div className="contact_item">
                            <label htmlFor="name">name</label><br/>
                            <input required type="text" id="name" name="name" placeholder="Nombre" onChange={this.handleInputChange} required/>
                        </div>
                        <div className={`contact_item  ${this.errorClass(this.state.formErrors.email)}`}>
                            <label htmlFor="email">Email</label><br/>
                            <input required type="email" id="email" name="email" placeholder="Email *" onChange={this.handleInputChange} required/>
                        </div>
                        <div className="contact_item">
                            <label className="inputSubject" htmlFor="inputSubject">Tema de Consulta</label><br/>
                            <select className="" id="inputSubject" name="subject" title="Choose subject" onChange={this.handleInputChange}>
                                <option value="">- Por favor, seleccione un tema de consulta -</option>
                                <option value="actividad">Info relativa a alguna actividad</option>
                                <option value="dpto">Contacta con nuestro dpto de actividades</option>
                                <option value="trabaja">Trabaja con nosotros</option>
                                <option value="sugerencias">Haznos sugerencias</option>
                                <option value="reclamaciones">Atendemos tus reclamaciones</option>
                                <option value="novedades">Te avisamos de nuestras novedades</option>
                                <option value="diferente">Algo diferente</option>
                            </select>
                        </div>
                        <div className="contact_item">
                            <label htmlFor="recomendation">Hace Aqui tu recomendaci√≥n</label><br/>
                            <textarea required className="contact_item" id="recomendation" name="message" placeholder="Introduzca aqui su mensaje *" onChange={this.handleInputChange} required></textarea>
                        </div>
                        <div className="contact_item"  disabled={!this.state.formValid}>
                            <input className="contact_Item" type="submit" name="submit" id="submit" value="Enviar" onClick={this.handleSubmit}/>
                        </div>
                    </form>
                </div> 
            </div>
        );
    }
}

export default Contact;