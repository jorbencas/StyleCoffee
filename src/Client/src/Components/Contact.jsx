import React from 'react';

class App extends React.Component {
    constructor(props){
        super(props);
        //T.setTexts(Settings.getTraductedText(), { MDFlavor: 0 });        
       // T.setTexts(require('../lib/i18n/' + Settings.getLanguage() + '.json'))
       /* this.state = {                
          motto:T.translate("motto")    
        }; */      
    }    
   /* componentDidMount() {
        this.subscription = events.subscribe('settings/change',(obj) => {  
            //T.setTexts(Settings.getTraductedText(), { MDFlavor: 0 })
            T.setTexts(require('../lib/i18n/' + Settings.getLanguage() + '.json'))
            this.setState({
                motto:T.translate("motto")     
            });               
        });      
    }

    componentWillUnmount() {
        this.subscription.remove();
    }
*/
    render() {               
        return (
            <div>
                 <p>Prueba</p>
            </div>
        );
    }
}

export default App;