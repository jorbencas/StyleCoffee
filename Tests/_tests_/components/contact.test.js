import React from 'react';
import {shallow, render} from 'enzyme';
import Contact from '../../../src/Client/Components/common/Contact';

it('renders no bookswhen store is empty', () => {
    const wrapper = shallow(<Contact/>)
    expect(wrapper).toMatchSnapshot();
});