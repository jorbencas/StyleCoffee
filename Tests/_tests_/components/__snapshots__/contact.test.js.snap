// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders no bookswhen store is empty 1`] = `
<div
  className="grid-main"
>
  <div
    className="Contact"
  >
    <form
      className="form-contact"
      id="contact_form"
      name="contact_form"
    >
      <h1
        className="contact_item"
      >
        Contacto
      </h1>
      <div>
        <FormErrors
          formErrors={
            Object {
              "email": "",
            }
          }
        />
      </div>
      <div
        className="contact_item"
      >
        <label
          htmlFor="name"
        >
          name
        </label>
        <br />
        <input
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          required={true}
          type="text"
        />
      </div>
      <div
        className="contact_item  Tu debes escribir algo"
      >
        <label
          htmlFor="email"
        >
          Email
        </label>
        <br />
        <input
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Email *"
          required={true}
          type="email"
        />
      </div>
      <div
        className="contact_item"
      >
        <label
          className="inputSubject"
          htmlFor="inputSubject"
        >
          Tema de Consulta
        </label>
        <br />
        <select
          className=""
          id="inputSubject"
          name="subject"
          onChange={[Function]}
          title="Choose subject"
        >
          <option
            value=""
          >
            - Por favor, seleccione un tema de consulta -
          </option>
          <option
            value="actividad"
          >
            Info relativa a alguna actividad
          </option>
          <option
            value="dpto"
          >
            Contacta con nuestro dpto de actividades
          </option>
          <option
            value="trabaja"
          >
            Trabaja con nosotros
          </option>
          <option
            value="sugerencias"
          >
            Haznos sugerencias
          </option>
          <option
            value="reclamaciones"
          >
            Atendemos tus reclamaciones
          </option>
          <option
            value="novedades"
          >
            Te avisamos de nuestras novedades
          </option>
          <option
            value="diferente"
          >
            Algo diferente
          </option>
        </select>
      </div>
      <div
        className="contact_item"
      >
        <label
          htmlFor="recomendation"
        >
          Hace Aqui tu recomendaci√≥n
        </label>
        <br />
        <textarea
          className="contact_item"
          id="recomendation"
          name="message"
          onChange={[Function]}
          placeholder="Introduzca aqui su mensaje *"
          required={true}
        />
      </div>
      <div
        className="contact_item"
        disabled={true}
      >
        <input
          className="contact_Item"
          id="submit"
          name="submit"
          onClick={[Function]}
          type="submit"
          value="Enviar"
        />
      </div>
    </form>
  </div>
</div>
`;
